clang/test/Parser/cxx0x-attributes.cpp

```diff
@@ -453,3 +453,12 @@ namespace P2361 {
 }
 
 alignas(int) struct AlignAsAttribute {}; // expected-error {{misplaced attributes; expected attributes here}}
+
+namespace GH108819 {
+void a([[maybe_unused]] void) {}                 // expected-error {{an attribute list cannot appear here}} \
```

Comment:
I think this is a confusing diagnostic -- an attribute list *can* appear here. The problem is that an attribute which appertains to a declaration cannot appertain to a void parameter. I think that should be handled in SemaDeclAttr.cpp when doing the usual appertainment checks.